app/src/main/java/so/codex/hawk/domain/providers/ProjectProvider.kt

```diff
@@ -0,0 +1,26 @@
+package so.codex.hawk.domain.providers
+
+import io.reactivex.rxjava3.core.Observable
+import io.reactivex.rxjava3.schedulers.Schedulers
+import io.reactivex.rxjava3.subjects.BehaviorSubject
+import so.codex.hawk.entity.Project
+
+object ProjectProvider {
+    private val subject: BehaviorSubject<List<Project>> = BehaviorSubject.create()
+
+    init {
+        WorkspaceProvider.getWorkspaces()
+            .observeOn(Schedulers.io())
+            .subscribe { workspaceList ->
+                subject.onNext(
+                    workspaceList.flatMap {
+                        it.projects
+                    }
```

Comment:
почему бы не сделать так?
```suggestion
            .observeOn(Schedulers.io())
            .map { workspaces ->
                workspaces.fold(listOf<Project>()) { acc, value ->
                    acc + value.projects
                }
            }
            .subscribe { projectList ->
                subject.onNext(projectList)
            }
```